lavalink/models.py

```diff
@@ -302,18 +304,17 @@ async def play(self, track: AudioTrack = None, start_time: int = 0, end_time: in
 
         options = {}
 
-        if start_time:
-            if 0 > start_time > track.duration:
+        if start_time is not None:
+            if not 0 <= start_time <= track.duration:
                 raise ValueError('start_time is either less than 0 or greater than the track\'s duration')
             options['startTime'] = start_time
 
-        if end_time:
-            if 0 > end_time > track.duration:
+        if end_time is not None:
+            if not 0 <= end_time <= track.duration:
                 raise ValueError('end_time is either less than 0 or greater than the track\'s duration')
             options['endTime'] = end_time
 
-        if no_replace:
-            options['noReplace'] = no_replace
+        options['noReplace'] = no_replace
```

Comment:
Perhaps the `if` check should remain, but instead should check if the given value is a boolean.

Reply:
How about raise ValueError if it isn't a bool?
```py
if not isinstance(no_replace, bool):
    raise ValueError('no_replace must be a bool')
options['noReplace'] = no_replace
```

Reply:
Unless it is intended behavior in the interface that the user can override the default `no_replace:bool = False` with a `None` but still get the functionality of `False`

Reply:
Either way should probably be explicit in what `noReplace` option we are requesting from Lavalink

Reply:
`TypeError` is the exception used for invalid parameter types.
I guess none could be accepted, so perhaps worth doing `no_replace = bool(...)` just to ensure Lavalink always receives what it expects.

Reply:
Oh yes, `TypeError` is definitely correct here, by bad.

My suggestion would be to not allow `None` as a valid input for a boolean parameter, especially when there is a default of `False` in the function sig.

However, if you think it is best to keep `None` as valid for legacy or anything else, then I don't think we should cast using `no_replace = bool(...)` because now suddenly `0` vs `1`, `''` vs `'true'`, and friends are also valid input. Also probably more clear to be explicit that `None` is equivalent to `False`.
```py
if no_replace is None:
    no_replace = False
if not isinstance(no_replace, bool):
    raise TypeError('no_replace must be a bool')
options['noReplace'] = no_replace
```